maven/poms.bzl

```diff
@@ -78,7 +108,68 @@ def _extract_dependencies(parent_node):
 # The parameter should be an xml node containing the tag <dependencyManagement>
 def _extract_dependency_management(project_node):
     node = xml.find_first(project_node, labels.DEPENDENCY_MANAGEMENT)
-    return poms.extract_dependencies(node) if bool(node) else []
+    return _extract_dependencies(node) if bool(node) else []
+
+def _merge_dependency(fallback, dependency):
+    if not bool(fallback):
+        return dependency
+
+    # If this were a node tree, we could re-use the leaf element merging, but it's a struct. :(
+    # groupid and artifactid are always present, so don't bother merging them.
+    return _dependency(
+        group_id = dependency.group_id,
+        artifact_id = dependency.artifact_id,
+        version = dependency.version if bool(dependency.version) else fallback.version,
+        type = dependency.type if bool(dependency.type) else fallback.type,
+        optional = dependency.optional if bool(dependency.optional) else fallback.optional,
+        scope = dependency.scope if bool(dependency.scope) else fallback.scope,
+        classifier = dependency.classifier if bool(dependency.classifier) else fallback.classifier,
+        system_path = dependency.system_path if bool(dependency.system_path) else fallback.system_path,
+    )
+
+def _get_variable(string):
+    start = string.find("${")
+    end = string.find("}")
+    return string[start + 2:end] if start >= 0 and end >= 0 and start < end - 3 else None
```

Comment:
🤣

Reply:
Strings are slices, don't know what to tell you.